pkg/apis/core/types.go

```diff
@@ -2787,6 +2787,22 @@ type Sysctl struct {
 	Value string
 }
 
+// PodFSGroupChangePolicy holds policies that will be used for applying fsGroup to a volume
+// when volume is mounted.
+type PodFSGroupChangePolicy string
+
+const (
+	// OnRootMismatch indicates that volume's ownership and permissions will be changed
+	// only when permission and ownership of root directory does not match with expected
+	// permissions on the volume. This can help shorten the time it takes to change
```

Comment:
we don't check permission match or not, only ownerhship here, right?

Reply:
no we check both. Not just permissions.

Reply:
We only check whether provided fsGroup and gid matches, and then skip changing ownership if they match. We don't check permission since user does not provide one from API, so there is no expected permission here.